src/data/list/zipper.lean

```diff
@@ -0,0 +1,183 @@
+/- © E.W.Ayers 2019 -/
+import control.traversable
+
+universes u v
+
+/-- A list zipper represents a list and a 'cursor' at a particular point in the list. -/
+@[derive decidable_eq]
+structure list.zipper (α : Type u) :=
+(left : list α)
+(right : list α)
```

Comment:
This is very similar to the current `string.iterator` I think - I wonder whether it can be de-duplicated.

Reply:
Good spot, I'll look in to it